### PR TITLE
ShareDrawer: Test coverage

### LINKED ISSUE #92839: ShareDrawer: Add tests
Although many components extend from already tested class components or reuse same logic, we need to create some tests to get better coverage

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

It adds tests for the new share drawer flow.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/92839

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: e2e/scenes/dashboards-suite/dashboard-export-json.spec.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            });
+          })
+          .then(() => {
+            cy.get('@url').then((url) => {
+              cy.wrap(url).should('include', '__inputs');
+            });
+          });
+      });
+
+    e2e.pages.ExportDashboardDrawer.ExportAsJson.cancelButton().click();
```
Comment (lucychen-grafana) [inline e2e/scenes/dashboards-suite/dashboard-export-json.spec.ts:69]: Should we add a test when Download File is clicked? Is it possible to check that it downloads?

File: e2e/scenes/dashboards-suite/dashboard-share-snapshot-create.spec.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        e2e.pages.Dashboard.Controls().should('exist');
+
+        // Validate the panels are rendered
+        for (const title of panelsToCheck) {
+          e2e.components.Panels.Panel.title(title).should('be.visible');
+        }
+      });
+  });
+});
+
```
Comment (lucychen-grafana) [inline e2e/scenes/dashboards-suite/dashboard-share-snapshot-create.spec.ts:64]: Should we add a test to delete and confirm delete?

File: e2e/scenes/dashboards-suite/dashboard-export-json.spec.ts (hunk 1 above)
Comment (juanicabanas) [reply e2e/scenes/dashboards-suite/dashboard-export-json.spec.ts:69]: I tried it and I didn't find an easy way, it's too complex. I think we're good with this right now

File: e2e/scenes/dashboards-suite/dashboard-share-snapshot-create.spec.ts (hunk 2 above)
Comment (juanicabanas) [reply e2e/scenes/dashboards-suite/dashboard-share-snapshot-create.spec.ts:64]: we can add it later. It covers the most important flow

**Review by lucychen-grafana — APPROVED** [review]
LGTM

**Pull request merged**